app/src/sharedTest/java/org/oppia/android/app/parser/CustomBulletSpanTest.kt

```diff
@@ -0,0 +1,186 @@
+import org.oppia.android.domain.classify.rules.numberwithunits.NumberWithUnitsRuleModule
+import org.oppia.android.domain.classify.rules.numericinput.NumericInputRuleModule
+import org.oppia.android.domain.classify.rules.ratioinput.RatioInputModule
+import org.oppia.android.domain.classify.rules.textinput.TextInputRuleModule
+import org.oppia.android.domain.onboarding.ExpirationMetaDataRetrieverModule
+import org.oppia.android.domain.oppialogger.LogStorageModule
+import org.oppia.android.domain.oppialogger.loguploader.LogUploadWorkerModule
+import org.oppia.android.domain.oppialogger.loguploader.WorkManagerConfigurationModule
+import org.oppia.android.domain.question.QuestionModule
+import org.oppia.android.domain.topic.PrimeTopicAssetsControllerModule
+import org.oppia.android.testing.TestAccessibilityModule
+import org.oppia.android.testing.TestDispatcherModule
+import org.oppia.android.testing.TestLogReportingModule
+import org.oppia.android.util.caching.testing.CachingTestModule
+import org.oppia.android.util.gcsresource.DefaultResourceBucketName
+import org.oppia.android.util.gcsresource.GcsResourceModule
+import org.oppia.android.util.logging.LoggerModule
+import org.oppia.android.util.logging.firebase.FirebaseLogUploaderModule
+import org.oppia.android.util.parser.CustomBulletSpan
+import org.oppia.android.util.parser.GlideImageLoaderModule
+import org.oppia.android.util.parser.HtmlParser
+import org.oppia.android.util.parser.HtmlParserEntityTypeModule
+import org.oppia.android.util.parser.ImageParsingModule
+import org.robolectric.annotation.Config
+import org.robolectric.annotation.LooperMode
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+@Config(application = CustomBulletSpanTest.TestApplication::class, qualifiers = "port-xxhdpi")
+class CustomBulletSpanTest {
+
+  private lateinit var launchedActivity: Activity
+
+  @Inject
+  lateinit var htmlParserFactory: HtmlParser.Factory
+
+  @Inject
+  @field:DefaultResourceBucketName
+  lateinit var resourceBucketName: String
+
+  @get:Rule
+  var activityTestRule: ActivityTestRule<HtmlParserTestActivity> = ActivityTestRule(
+    HtmlParserTestActivity::class.java, /* initialTouchMode= */ true, /* launchActivity= */ false
+  )
+
+  @Before
+  fun setUp() {
+    setUpTestApplicationComponent()
+    Intents.init()
+    val intent = Intent(Intent.ACTION_PICK)
+    launchedActivity = activityTestRule.launchActivity(intent)
+  }
+
+  @After
+  fun tearDown() {
+    Intents.release()
+  }
+
+  @Test
+  fun testCustomBulletSpan_hasCorrectlySpacedLeadingMargin() {
+    val textView =
+      activityTestRule.activity.findViewById(R.id.test_html_content_text_view) as TextView
+    val htmlParser = htmlParserFactory.create(
+      resourceBucketName,
+      /* entityType= */ "",
+      /* entityId= */ "",
+      /* imageCenterAlign= */ true
+    )
+    val htmlResult: Spannable = htmlParser.parseOppiaHtml(
```

Comment:
So the purpose of this test is to specifically verify CustomBulletSpan, not that HtmlParser uses CustomBulletSpan. To make this a bit simpler, I suggest moving the following block of code to actually replace bullet spans with the custom one in a spannable to a companion function of the bullet span class:

https://github.com/oppia/oppia-android/blob/develop/utility/src/main/java/org/oppia/android/util/parser/HtmlParser.kt#L82

E.g.:

```kotlin
class CustomBulletSpan {
  companion object {
    fun replaceBulletSpans(spannableBuilder: SpannableStringBuilder): SpannableStringBuilder {
      ...
    }
  }
}
```

And then, we can call that function both in HtmlParser & here. The latter (calling it here) simplifies the test, and it verifies just CustomBulletSpan rather than the span in the context of HtmlParser.

Reply:
@FareesHussain Please reply to this.

Reply:
Done